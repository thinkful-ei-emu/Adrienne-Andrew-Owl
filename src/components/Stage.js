import React from 'react';
import Participant from './Participant';

export default function Stage(props) {

  if (!props.participants) {
    return (
      <div></div>
    )
  }

  const staged = props.participants.filter(participant => participant.onStage);
  const participants = staged.map(participant => {
    return <Participant
      key = {participant.id}
      name={participant.name}
      avatar={participant.avatar}
      inSession={participant.inSession}
      onStage={participant.onStage}
      />
  });
  
  return (
    <div className="stage">
      {participants}
    </div>
  )
}