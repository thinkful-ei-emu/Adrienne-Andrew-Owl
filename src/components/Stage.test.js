import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Stage from './Stage';
import participants from '../data/participantData';

// smoke test; failing time
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Stage
    participants={[]}
    />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// screenshot test;
it('renders this UI as expected', () => {
  const appTest = renderer.create(<Stage
    participants={participants}
    />).toJSON();
  // Check whether it matches the previous snapshot
  // Stored in __snapshots__/App.test.js.snap
  expect(appTest).toMatchSnapshot(); 
});