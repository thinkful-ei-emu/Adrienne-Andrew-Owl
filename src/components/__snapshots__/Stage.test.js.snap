// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders this UI as expected 1`] = `
<div
  className="stage"
>
  <div
    className="participant"
  >
    <img
      alt="Avatar Here but no Avatar, sucks"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <p>
      Koren Templeton
    </p>
    <p>
      
    </p>
    <p>
      On Stage
    </p>
  </div>
  <div
    className="participant"
  >
    <img
      alt="Avatar Here but no Avatar, sucks"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <p>
      Ashla Attwool
    </p>
    <p>
      
    </p>
    <p>
      On Stage
    </p>
  </div>
</div>
`;
